numba/core/compiler_machinery.py

```diff
@@ -355,7 +355,7 @@ def run(self, state):
                 if isinstance(pass_inst, CompilerPass):
                     self._runPass(idx, pass_inst, state)
                 else:
-                    raise BaseException("Legacy pass in use")
+                    raise RuntimeError("Legacy pass in use")
```

Comment:
Suggest to undo this change.
The use of `BaseException` is to make it not catchable by `except Exception`.

Reply:
Fixed in 6e47aa2